### ISSUE TITLE
Rewrite bails-wallet in python

### ISSUE THREAD

**Issue by BenWestgate** [BenWestgate/Bails#4]
`bails-wallet` is longer than expected. The user interface is excellent and bug-free, however the code is not as easy to understand as it could be. The functions especially are not logically grouped or ordered making it more difficult to follow along.
As it is the most security critical file in the project it should have great readability.

It may be helpful to:
1. split it into multiple files based on logical components or functionality.
2. group and sort the functions logically
3. move unused functions for non-mvp features to a new file

**Comment by BenWestgate** [comment]
I started a refactor and organized the functions by their earliest usage in the script.

Functions used only once were removed. Unused variables removed.

There's still a 100 line function that may be better as a library called by bails-wallet as it exclusively processes text, calls no function and returns 0 or 1.

A consistent style to function declarations would benefit.

Postpone the full refactor until L1 is feature complete as that will determine whether more functions used only once can be merged or not.

**Comment by BenWestgate** [comment]
As mentioned in #5 
`bails-wallet` should be able to run stand-alone by any linux system with bitcoin core.

With start-up parameters or environment variables telling it where to look for blockchain data and other resources.

Fork it to a new repository if necessary.

 into it's own repo so that any Bitcoin Core user on Linux can create and restore codex32 backups.

**Comment by BenWestgate** [comment]
This unfortunately depends on two pull requests I'm involved with from Blockstream Research.

A considerable (up to all of it) could be converted to python. But I will choose the max readability and maintainability gains per time investment.

**Comment by BenWestgate** [comment]
It has been decreed that bails-wallet will become a stand-alone cross platform python package. So the refactor is occurring as it becomes a python package.

It may even be possible to maintain 3 repositories bitcoin-core-on-tails just installs bitcoin core and shortcuts for it, nothing else.

bails runs bitcoin-core-on-tails and installs and runs bails-wallet plus installs bails menu

bails-wallet runs the wallet code on any platform with python and bitcoin core.

**Comment by BenWestgate** [comment]
The rewrite will be complete and the bash `bails-wallet` removed when the L1 features have been tested and working in https://github.com/BenWestgate/Bails-Wallet

**Comment by BenWestgate** [comment]
> It may even be possible to maintain 3 repositories bitcoin-core-on-tails just installs bitcoin core and shortcuts for it, nothing else.
> 
> 
> 
> bails runs bitcoin-core-on-tails and installs and runs bails-wallet plus installs bails menu
> 
> 
> 
> bails-wallet runs the wallet code on any platform with python and bitcoin core.

@epiccurious do you support moving install-core and some of its supporting files to my "bitcoin-core-on-tails" repo? Then b could clone bitcoin-core-on-tails as a dependency.

It would have a readme with install instructions like Bails but the final result would just be Bitcoin core installed on Tails. Without any extra features beyond what you'd get installing Bitcoin core on Ubuntu from the snap store.

I think that could help close "Spooky Integration"

**Comment by BenWestgate** [comment]
The same sort of minimum setup could be a launch parameter or a different installation script but I'm unsure the technical level of the issue opener would be satisfied by that and it would require filtering the files to copy or splitting them into 2 directories in this repo i.e. bails and bitcoin-core-on-tails

**Comment by BenWestgate** [comment]
Actively being worked on.
